osu.Server.Spectator/Hubs/StatefulUserHub.cs

```diff
@@ -35,8 +35,18 @@ public override async Task OnConnectedAsync()
         {
             Console.WriteLine($"User {CurrentContextUserId} connected!");
 
-            // if a previous connection is still present for the current user, we need to clean it up.
-            await cleanUpState(false);
+            try
+            {
+                // if a previous connection is still present for the current user, we need to clean it up.
+                await cleanUpState(false);
+            }
+            catch
+            {
+                // if any exception happened during clean-up, don't allow the user to reconnect.
+                // this limits damage to the user in a bad state if their clean-up cannot occur (they will not be able to reconnect until the issue is resolved).
+                Context.Abort();
+                throw;
```

Comment:
Think #30 might be rendered obsolete/unnecessary with this change? I'm not yet totally sold on merging checks like the one in that PR without at least partial understanding of what the mechanism of failure is there, as it might turn out we'll be patching such cases until the end of time because there's a larger underlying problem - and this seems like it could be a plausible explanation.

Reply:
#30 is still required. it's still getting hit with both deployed, and even if it wasn't, i'd still want it in place as per the comment (the more failsafes the better; primary goal is to keep as much of the server working for as long as possible, isolating any issues)

this is how i always design server components: protect and then fix.